scene/gui/rich_text_label.cpp

```diff
@@ -3611,6 +3611,72 @@ bool RichTextLabel::is_selection_enabled() const {
 	return selection.enabled;
 }
 
+int RichTextLabel::get_selection_from() const {
+	if (!selection.active || !selection.enabled) {
+		return -1;
+	}
+
+	int index_from = 0;
+
+	RichTextLabel::Item *item = selection.from_item;
+
+	while (item) {
+		if (item->type == ITEM_TEXT) {
+			if (item == selection.from_item) {
+				index_from += selection.from_char;
+			} else {
+				const String &text = static_cast<ItemText *>(item)->text;
+				index_from += text.length();
+			}
+		} else if (item->type == ITEM_NEWLINE) {
+			if (item != selection.from_item) {
+				index_from++;
+			}
+		} else if (item->type == ITEM_INDENT) {
+			if (item != selection.from_item) {
+				index_from++;
+			}
+		}
+
+		item = _get_prev_item(item, true);
+	}
+
+	return index_from;
+}
+
+int RichTextLabel::get_selection_to() const {
+	if (!selection.active || !selection.enabled) {
+		return -1;
+	}
+
```

Comment:
And here just this:
`return selection.to_frame->lines[selection.to_line].char_offset + selection.to_char;`

Reply:
I've changed this one as well, but with `- 1` at the end due to the fact this method returns the last character for the selection.